core/types/hashing.go

```diff
@@ -97,6 +98,9 @@ func DeriveSha(list DerivableList, hasher TrieHasher) common.Hash {
 	for i := 1; i < list.Len() && i <= 0x7f; i++ {
 		indexBuf = rlp.AppendUint64(indexBuf[:0], uint64(i))
 		value := encodeForDerive(list, i, valueBuf)
+		if len(value) == 0 {
```

Comment:
This shouldn't be a fatal, probably a debug log

Reply:
Changed to error, this is a consensus-breaking error if it happens

Reply:
can you explain when this happens again? or have a testcase to demonstrate that

Reply:
It happens if elements in the list are nil